### PR TITLE
paper only event `player granted advancement criterion` 

### PR DESCRIPTION
**Pull request by BreadcrumbIsTaken**
### `player granted advancement criterion`

Fires when the player is granted an advancement criterion for an advancement. 

#### Switches
- `advancement:<name>` - the key of the advancement
- `criterion:<name>` - the criterion needed

#### Contexts
- `<context.advancement>` - the key of the advancement
- `<context.criterion>` - the criterion granted

Works on versions *1.17.1*, *1.18.2*, and *1.19.3*

Requested by Mergu on discord

###### 🎈 🎈 🎈

### CODE REVIEW

File: paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerGrantedAdvancementCriterionScriptEvent.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    // @Group Paper
+    //
+    // @Plugin Paper
+    //
+    // @Cancellable true
+    //
+    // @Switch advancement:<name> to only fire if the advancement for the criterion has the specified name.
+    // @Switch criterion:<name> to only fire if the criterion being granted has the specified name.
+    //
+    // @Triggers when a player is granted advancement criterion for an advancement.
```
Comment (mcmonkey4eva) [inline paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerGrantedAdvancementCriterionScriptEvent.java]: English is wonky here

File: paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerGrantedAdvancementCriterionScriptEvent.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    // @Example
+    // # This can narrate something like:
+    // # "Good job! You completed some criteria for the advancement: minecraft:story/root!"
+    // on player granted advancement criterion:
+    // - narrate "Good job! You completed some criteria for the advancement: <context.advancement>!"
+    //
+    // @Example
+    // # This will only narrate when the player is granted the criterion for taming a Calico cat
+    // # for the "A Complete Catalogue" advancement.
+    // on player granted advancement criterion advancement:minecraft:husbandry/complete_catalogue criterion:minecraft:calico:
```
Comment (mcmonkey4eva) [inline paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerGrantedAdvancementCriterionScriptEvent.java]: This is a really awkward line. Should make `criterion:calico` work.

**Comment by BreadcrumbIsTaken** [comment]
Pushed changes :)

File: paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerGrantedAdvancementCriterionScriptEvent.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    }
+
+    public ElementTag criterion;
+    public ElementTag advancement;
+    public PlayerAdvancementCriterionGrantEvent event;
+
+    @Override
+    public boolean matches(ScriptPath path) {
+        NamespacedKey advancementKey = Utilities.parseNamespacedKey(path.switches.get("advancement"));
+        if (!advancementKey.toString().equals(advancement.toString())) {
```
Comment (mcmonkey4eva) [inline paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerGrantedAdvancementCriterionScriptEvent.java]: Now this doesn't allow advanced matchers anymore?

**Comment by BreadcrumbIsTaken** [comment]
Done :>

**Pull request merged**